assets/src/BlockEditorValidation.js

```diff
@@ -0,0 +1,93 @@
+const { registerPlugin } = wp.plugins;
+const { PluginPrePublishPanel } = wp.editPost;
+const { dispatch } = wp.data;
+const {__} = wp.i18n;
+
+export const blockEditorValidation = () => {
+  registerPlugin( 'pre-publish-checklist', { render: PrePublishCheckList } );
+};
+
+const PrePublishCheckList = () => {
+  let lockPost = false;
+
+  let checkListMsg = '';
+  let classname    = '';
+
+  // "Analytics & Tracking" fields validation.
+  let $htmlbody   = $( 'html, body' );
+  // Filter CMB2 fields that have the 'data-validation' attribute set to 'required'.
+  let $toValidate = $( '[data-validation]' );
+
+  let analyticsErrorFlag   = false;
+
+  if ( $toValidate.length ) {
+    let $row = null;
+
+    $toValidate.each(function () {
+      const $this = $(this);
+      const val = $this.val();
+      $row = $this.parents('.cmb-row');
+
+      if ($this.is('[type="button"]') || $this.is('.cmb2-upload-file-id')) {
+        return true;
+      }
+
+      // Apply validation only for campaign post types.
+      if ('campaign' === $('#post_type').val() && 'required' === $this.data('validation')) {
+        if ($row.is('.cmb-type-file-list')) {
+
+          const has_LIs = $row.find('ul.cmb-attach-list li').length > 0;
+
+          if (!has_LIs) {
+            $row.find('select').addClass('cmb2_required_field_error');
```

Comment:
You can extract whether the row is valid into a const before changing the class. That way you can do that with a single call to `toggleClass` (passing the const) instead of 4 calls to `addClass` and `removeClass`.